### PR TITLE
Add configurable op_type for index watcher action

### LINKED ISSUE #64535: [Watcher] Index action can not write to data streams
**Elasticsearch version** (`bin/elasticsearch --version`):
7.9.1

**Description of the problem including expected versus actual behavior**:
The Watcher index action is not able to write to data streams due to using the op_type `index`, and not `create`. Watcher should be able to create documents in data streams.

**Steps to reproduce**:

 1. Create datasteam
 2. Create Watcher with index action 
 3. Execute Watcher

```json
PUT /_data_stream/logs-data-stream


PUT _watcher/watch/log_error_watch
{
  "trigger": {
    "schedule": {
      "interval": "100000h"
    }
  },
  "input": {
    "search": {
      "request": {
        "indices": [
          "logs"
        ],
        "body": {
          "query": {
            "match_all": {}
          }
        }
      }
    }
  },
  "actions": {
    "index_payload": {
      "index": {
        "index": "logs-data-stream",
        "doc_id": "my-id"
      }
    }
  }
}

POST _watcher/watch/log_error_watch/_execute

DELETE _watcher/watch/log_error_watch

DELETE _data_stream/logs-data-stream
```

**Provide logs (if relevant)**:

```
"actions" : [
        {
          "id" : "index_payload",
          "type" : "index",
          "status" : "failure",
          "error" : {
            "root_cause" : [
              {
                "type" : "illegal_argument_exception",
                "reason" : "only write ops with an op_type of create are allowed in data streams"
              }
            ],
            "type" : "illegal_argument_exception",
            "reason" : "only write ops with an op_type of create are allowed in data streams"
          }
        }
      ]
```

### PR DESCRIPTION
**Pull request by danhermann**
This allows the index action to work with data streams which require an `op_type` of `create`.

Fixes #64535

### CODE REVIEW

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java [hunk 1]
```diff
@@ -143,6 +156,13 @@ public static IndexAction parse(String watchId, String actionId, XContentParser
             } else if (token == XContentParser.Token.VALUE_STRING) {
                 if (Field.DOC_ID.match(currentFieldName, parser.getDeprecationHandler())) {
                     docId = parser.text();
+                } else if (Field.OP_TYPE.match(currentFieldName, parser.getDeprecationHandler())) {
+                    try {
+                        opType = DocWriteRequest.OpType.fromString(parser.text());
```
Comment (martijnvg) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java:161]: maybe validate that only create or index op type can be used?

**Review by martijnvg — APPROVED** [review]
Maybe also document that if the `index` action should be used with a data stream that the optype must be set to create?
LGTM

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java (hunk 1 above)
Comment (danhermann) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java:162]: Thanks, @martijnvg. What versions do you think we should target for this?

Comment (martijnvg) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java:162]: 8.0.0 and 7.11?

**Pull request merged**